cmake/bintools/arcmwdt/target.cmake

```diff
@@ -22,6 +22,18 @@ find_program(CMAKE_GDB     ${CROSS_COMPILE}mdb     PATHS ${TOOLCHAIN_HOME} NO_DE
 
 # MWDT binutils don't support required features like section renaming, so we
 # temporarily had to use GNU objcopy instead
+
+# ZEPHYR_SDK_CROSS_COMPILE was defined earlier for DTC preprocessing while ARCH variable
+# was undefined. Redefine it with correct value. Save and restore values affected by target.cmake.
+set(save_CROSS_COMPILE CROSS_COMPILE)
+set(save_SYSROOT_DIR SYSROOT_DIR)
+include(${ZEPHYR_SDK_INSTALL_DIR}/cmake/zephyr/target.cmake)
```

Comment:
why is `target.cmake` from Zephyr SDK needed ?
I was under the impression that the only reason for loading Zephyr SDK into arcmwdt which is now done when loading generic.cmake.

Reply:
Hi @tejlmand ,
Indeed, here we make redefinition of  ZEPHYR_SDK_CROSS_COMPILE variable, that originally was defined with some default value. Previous definition was incorrect because on early stage ARCH variable was undefined. But at the same time even incorrect ZEPHYR_SDK_CROSS_COMPILE is quite enough to make DTC works.
And yes, it's ugly workaround, but I was unable to find appropriate solution.

Reply:
> Here, on this stage objcopy with valid ARCH-dependent prefix should be used.

So is my understanding correct that arcmwdt requires objcopy from Zephyr SDK ?
(Does it require other tools as well ?)

If that is the case, then that is a very important piece of information and should thus be described in the commit message (and a code comment would be nice as well).

Will take a look at the code which this piece of information is mind.

Reply:
Original code already contains such information (https://github.com/kokas-a/zephyr/blob/4b276c496201c27abfde4cd8cff2b3d56c8a08bf/cmake/bintools/arcmwdt/target.cmake#L23). It was added long before this commit.

Reply:
> It was added long before this commit, and I think it is not necessary duplicate it one more time.

Right, `objcopy` is mentioned above. Unsure how I missed that.

Loading of `target.cmake` here thus makes sense, but to minimize the risk of leaking variables from Zephyr SDK's target.cmake then loading should be wrapped in a function.
This also means that the store/restore of settings are no longer needed because a function will scope variables.

Like this:
```
function(zephyr_sdk_target_cmake)
  # Scoped loading of variables set by Zephyr SDK target.cmake.
  include(${ZEPHYR_SDK_INSTALL_DIR}/cmake/zephyr/target.cmake)

  # MWDT binutils don't support required features like section renaming, so we
  # temporarily had to use GNU objcopy instead
  find_program(CMAKE_OBJCOPY ${CROSS_COMPILE}objcopy PATHS ${ZEPHYR_SDK_INSTALL_DIR} NO_DEFAULT_PATH)
  message(STATUS "Found GNU objcopy helper for MWDT: ${CMAKE_OBJCOPY} (Zephyr SDK ${SDK_VERSION})")
endfunction()

zephyr_sdk_target_cmake()
```
and then all the code related to `set(save_CROSS_COMPILE ...)`, `unset(CROSS_COMPILE)`, etc can be removed.